### ISSUE TITLE
Enabled OCS Share API CORS

### ISSUE THREAD

**Issue by Silbad** [digital-blueprint/webapppassword#1]
Hello,
Can you add the possibility to enabled CORS pour OCS Share API (https://docs.nextcloud.com/server/latest/developer_manual/client_apis/OCS/ocs-share-api.html) ? Webdav don't support sharing links...
Thanks ^^

**Comment by pbek** [comment]
Thank you for your suggestion.
First someone would need to find out what CORS headers are supposed to be set and that headers shouldn't be on by default. So there would need to be more configuration options on the admin page.

**Comment by vialcollet** [comment]
Hi
This would be really nice to have this feature.
I am creating an obisdian plugin to share notes but I am blocked by this CORS issue.

**Comment by aleixq** [comment]
I am looking too at how this could be solved... 

Adding the annotation `@CORS`  createShare method in ShareAPIController.php of files_sharing app , and adding the route in appinfo/routes.php:

 ```
[                                       
 'name' => 'ShareAPI#preflighted_cors',
 'url' => '/api/v1/shares',
 'verb' => 'OPTIONS',
],
```

allows the sharing api to be used  from an external url to create a share (not a surprise as it's said in https://docs.nextcloud.com/server/22/developer_manual/digging_deeper/rest_apis.html#rest-apis  ).

Altering the core files_sharing app isn't the way to go, but I cannot find any event to listen and react, to adapt the response to OPTIONS  and POST method request to /api/v1/shares files_share route. This would help when using the api in a dedicated app ... 

Something like https://symfony.com/doc/current/reference/events.html#kernel-response would be great  But using symfony events is discouraged

I'll keep looking.

**Comment by pbek** [comment]
Nice. If you want to integrate it into webapppassword, I can review the pull request.

**Comment by pbek** [comment]
## 23.4.0
- A files sharing API with origin check was added
  (for [#1](https://github.com/digital-blueprint/webapppassword/issues/1), thank you @aleixq)

**Comment by pbek** [comment]
I released it on https://apps.nextcloud.com/apps/webapppassword as well.

**Comment by aleixq** [comment]
Just adding that a PR is being proposed in: https://github.com/nextcloud/server/pull/37896 and in related issue.
